static_frame/core/frame.py

```diff
@@ -1514,21 +1516,36 @@ def from_sql(cls,
             if index_depth == 1:
                 index_constructor = Index
 
-                def row_gen_final() -> tp.Iterator[tp.Sequence[tp.Any]]:
-                    for row in row_gen:
+                def row_gen() -> tp.Iterator[tp.Sequence[tp.Any]]:
+                    for row in sql_iterable:
                         index.append(row[0])
                         yield row[1:]
 
             else: # > 1
                 index_constructor = IndexHierarchy.from_labels
 
-                def row_gen_final() -> tp.Iterator[tp.Sequence[tp.Any]]:
-                    for row in row_gen:
+                def row_gen() -> tp.Iterator[tp.Sequence[tp.Any]]:
+                    for row in sql_iterable:
                         index.append(row[:index_depth])
                         yield row[index_depth:]
         else:
             index = None
-            row_gen_final = lambda: row_gen
+            row_gen = lambda: sql_iterable
+
+        if columns_select:
+            columns_select = set(columns_select)
+
+            sel = columns.isin(columns_select)
+            # desired_cols = columns.loc_to_iloc(sel) # Currently doesn't work for IH...
+            desired_cols = np.where(sel)[0]
+
+            def row_gen_final() -> tp.Iterator[tp.Sequence[tp.Any]]:
+                for row in row_gen():
+                    yield itemgetter(*desired_cols)(row)
```

Comment:
I like this use of `itemgetter`! However, can we call `itemgetter(*desired_cols)` once, before the `for` loop, and then simply re-use it for each row?

Reply:
I didn't think of that - will change.